### PR TITLE
python,pythonPackages: make aliases

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/issues/148779

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zimbatm** [comment]
Result of `nixpkgs-review pr 154687` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>libsForQt512.kalzium</li>
    <li>libsForQt514.kalzium</li>
    <li>python310Packages.tiledb</li>
    <li>python39Packages.tiledb</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>ddar</li>
  </ul>
</details>
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>avogadro2</li>
    <li>cloudcompare</li>
    <li>libsForQt5.kalzium (libsForQt515.kalzium ,plasma5Packages.kalzium)</li>
    <li>lumo</li>
    <li>openbabel (openbabel3)</li>
    <li>pdal</li>
    <li>roundup</li>
    <li>rsyslog</li>
    <li>rsyslog-light</li>
    <li>schismtracker</li>
    <li>seasocks</li>
    <li>shocco</li>
    <li>spacevim</li>
    <li>tebreak</li>
    <li>tests.vim.test_vim_with_vim_nix</li>
    <li>tiledb</li>
  </ul>
</details>

**Comment by danielbarter** [comment]
just to make sure i understand, this is just changing `python` to `python2` in situations where python2 was being passed in as an argument at top level?

**Comment by dotlambda** [comment]
> just to make sure i understand, this is just changing `python` to `python2` in situations where python2 was being passed in as an argument at top level?

No, only where neither `python2` nor `python3` were used explicitly. If there is something like `override { python = python2; }` in `all-packages.nix` then this PR doesn't touch that package.

**Review by danielbarter — APPROVED** [review]
cool, looks fine to me. Would be good to get some more experienced opinions though

**Comment by dotlambda** [comment]
> Would be good to get some more experienced opinions though

I mostly requested your review because I made some package that you maintain use Python 3 instead of Python 2.

**Comment by danielbarter** [comment]
Oh, interesting. You are talking about openbabel? I looked for that in the diff but couldn't see it. I use openbabel with python3 anyway. Wasn't aware that it was defaulting to python2

**Comment by danielbarter** [comment]
ahh, i see. That python2 version was depended on by something else, which is why we spun it out. I never actually use it

**Review by FredeHoey — APPROVED** [review]
Seasocks looks good 👍

**Comment by FRidh** [comment]
Correct me if I am wrong, but you removed the attribute, and then went through all the eval errors and test whether passing in `python3` worked?

This is a follow-up to https://github.com/NixOS/nixpkgs/pull/117570.

**Comment by dotlambda** [comment]
> Correct me if I am wrong, but you removed the attribute, and then went through all the eval errors and test whether passing in `python3` worked?

I did remove the attribute and went through the eval errors but only checked whether python3 works for very few of them.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -20189,9 +20184,7 @@ with pkgs;
     stdenv = gcc6Stdenv;
   });
 
-  v8 = callPackage ../development/libraries/v8 {
-    inherit (python2Packages) python;
-  };
+  v8 = callPackage ../development/libraries/v8 { };
```
Comment (FRidh) [inline pkgs/top-level/all-packages.nix:20187]: rebase needed because of https://github.com/NixOS/nixpkgs/pull/154007

**Review by FRidh — APPROVED** [review]
While ideally we'd go through all of these packages as well, these are now leaf packages, which the individual maintainers can take care of themselves.

**Comment by dotlambda** [comment]
> While ideally we'd go through all of these packages as well, these are now leaf packages, which the individual maintainers can take care of themselves.

I'm in the process of doing that, marking as a draft for now.

**Comment by dotlambda** [comment]
Actually I won't have time to go through all of them. We should rather ping their maintainers in https://github.com/NixOS/nixpkgs/issues/148779.
cc @piegamesde

**Comment by piegamesde** [comment]
I'll be in a similar situation with #155061, and my decision for that was to track the maintainers of the packages I touch as I go and then ping or review request everybody once its ready. What do you think?

**Comment by FRidh** [comment]
Thanks @dotlambda

**Pull request merged**